src/main/java/io/github/bobocodebreskul/config/PropertiesConfiguration.java

```diff
@@ -0,0 +1,105 @@
+package io.github.bobocodebreskul.config;
+
+import static java.util.Objects.isNull;
+
+import io.github.bobocodebreskul.context.exception.ConfigurationFileNotFoundException;
+import io.github.bobocodebreskul.context.exception.InvalidPropertyValueException;
+import io.github.bobocodebreskul.context.exception.LoadingPropertiesFailedException;
+import java.io.InputStream;
+import java.util.Properties;
+import lombok.extern.slf4j.Slf4j;
+
+/**
+ * PropertiesConfiguration class stores properties from configuration file and provides an API to
+ * get property as string or integer value.
+ */
+@Slf4j
+public class PropertiesConfiguration {
+
+  private static String CONFIG_FILE = "application.properties";
+
+  private static Properties properties;
```

Comment:
```suggestion
  private static Properties properties = new Properties();
```
so that you will avoid NPE, once you forgot to call 'loadProperties' method before retrieving the properties